src/Web/Spock/Internal/Cookies.hs

```diff
@@ -0,0 +1,80 @@
+{-# LANGUAGE CPP               #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RecordWildCards   #-}
+
+module Web.Spock.Internal.Cookies
+    ( CookieSettings(..)
+    , defaultCookieSettings
+    , CookieEOL(..)
+    , generateCookieHeaderString
+    )
+where
+
+import           Control.Monad.IO.Class (liftIO)
+import           Data.Monoid            ((<>))
+import qualified Data.Text              as T
+import           Data.Time
+#if MIN_VERSION_time(1,5,0)
+#else
+import           System.Locale          (defaultTimeLocale)
+#endif
+
+data CookieSettings = CookieSettings { csEOL      :: CookieEOL
+                                     , csHTTPOnly :: Bool
+                                     , csSecure   :: Bool
+                                     , csDomain   :: T.Text
+                                     , csPath     :: T.Text
+                                     }
+
+data CookieEOL = CookieValidUntil UTCTime
+               | CookieValidFor NominalDiffTime
+               | CookieValidForSession
+
+defaultCookieSettings :: CookieSettings
+defaultCookieSettings = CookieSettings { csEOL      = CookieValidForSession
+                                       , csHTTPOnly = False
+                                       , csSecure   = False
+                                       , csDomain   = T.empty
+                                       , csPath     = "/"
+                                       }
+
+generateCookieHeaderString :: T.Text -> T.Text -> CookieSettings -> IO T.Text
+generateCookieHeaderString name value CookieSettings{..} = do
```

Comment:
I move the `do` into the next line in Spock, would be great if you could follow that, too.